### PR TITLE
[forward-port] Only keep static members when looking up java static members

### PR DESCRIPTION
**Pull request by lrytz**
Forward port https://github.com/scala/scala/pull/10045

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/TreeGen.scala [hunk 1]
```diff
@@ -174,7 +174,7 @@ abstract class TreeGen extends scala.reflect.internal.TreeGen with TreeDSL {
    */
   def convertToTypeName(tree: Tree): Option[RefTree] = tree match {
     case Select(qual, name) => Some(Select(qual, name.toTypeName))
-    case Ident(name)        => Some(Ident(name.toTypeName).copyAttrs(tree))
+    case Ident(name)        => Some(Ident(name.toTypeName))
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/ast/TreeGen.scala:177]: @som-snytt do you have memories about https://github.com/scala/scala/commit/4f40dc87e4 ?

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/TreeGen.scala:177]: No, I'll look. I have a branch `save/5715-extra-mechanism` so probably I was experimenting.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/TreeGen.scala:177]: March 3 is when you told me about `// javaVersion` so I still owe you one from that week.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/TreeGen.scala:177]: It seems to have been needed for the "first take" described at https://github.com/scala/bug/issues/5715#issuecomment-1059776386  where (guessing a bit) in
```
new `then`
```
the ident loses the attachment that says it was backquoted. Maybe that only matters post-typer for tree printing now.

Comment (som-snytt) [reply src/compiler/scala/tools/nsc/ast/TreeGen.scala:177]: Yes, except the backquotes are not printed at parser. Sorry for the detour.

My compensatory pun is: That was `then`, this is now.

**Review by som-snytt — APPROVED** [review]
LGTM forward-porting retronym approval

**Comment by lrytz** [comment]
Thanks!

**Comment by lrytz** [comment]
This seems to have broken the build on JDKs > 8 (https://github.com/scala/scala/actions/runs/2554347003)

**Comment by lrytz** [comment]
https://github.com/scala/scala/pull/10059

**Pull request merged**